arrow-fx-stm/src/main/kotlin/arrow/fx/stm/TSem.kt

```diff
@@ -1,20 +1,98 @@
 package arrow.fx.stm
 
-import arrow.fx.coroutines.Semaphore
-
 fun STM.newTSem(initial: Int): TSem = TSem(newTVar(checkNotNegative(initial)))
 
 /**
- * [TSem] is the transactional analog to [Semaphore].
+ * [TSem] is the transactional Semaphore.
  *
- * Semaphores are mostly used to limit concurrent access to resources and have to major operations:
- * - [STM.acquire] to acquire 1 or n permits
- * - [STM.release] to release 1 or n permits
+ * Semaphores are mostly used to limit concurrent access to resources by how many permits it can give out.
  *
  * ## Creating a [TSem]
  *
  * A [TSem] is created by using either [TSem.new] outside of transactions or [STM.newTSem] inside a transaction.
  * Both of these methods throw if the supplied initial value is negative.
+ *
+ * ## Acquiring one or more permits
+ *
+ * ```kotlin:ank:playground
+ * import arrow.fx.stm.TSem
```

Comment:
Can these be named TSemaphore or similar instead of the abbreviated names? I think for var is well known but the rest are not descriptive and can be confusing to users.